packages/library/src/commonMain/kotlin/io/realm/RealmConfiguration.kt

```diff
@@ -17,52 +17,54 @@
 package io.realm
 
 import io.realm.internal.Mediator
+import io.realm.internal.RealmModelInternal
+import io.realm.internal.RealmObjectCompanion
 import io.realm.interop.NativePointer
 import io.realm.interop.RealmInterop
 import io.realm.interop.SchemaMode
-import io.realm.interop.Table
+import kotlin.reflect.KClass
 
-class RealmConfiguration private constructor(
-    val path: String?, // Full path if we don't want to use the default location
-    val name: String?, // Optional Realm name (default is 'default')
-    val schema: Mediator,
-    val version: Long = 0,
-    val tables: List<Table> = listOf()
+class RealmConfiguration(
+    val path: String? = null, // Full path if we don't want to use the default location
+    val name: String = "default", // Optional Realm name (default is 'default')
+    val schema: List<KClass<out RealmObject>> // classes literal (T::class) //TODO rename ot model?
 ) {
-
+    internal var mapOfKClassWithCompanion: Map<KClass<*>, RealmObjectCompanion> = emptyMap()
     internal val nativeConfig: NativePointer = RealmInterop.realm_config_new()
+    internal lateinit var mediator: Mediator
+
+    // called by the compiler plugin, with a populated companion map
+    internal constructor (path: String?, name: String = "default", companionMap: Map<KClass<*>, RealmObjectCompanion>) :
+        this(path, name, emptyList()) {
+            mapOfKClassWithCompanion = companionMap
+            init()
+        }
 
-    init {
-        RealmInterop.realm_config_set_path(nativeConfig, path!!)
-        RealmInterop.realm_config_set_schema_mode(nativeConfig, SchemaMode.RLM_SCHEMA_MODE_AUTOMATIC)
-        RealmInterop.realm_config_set_schema_version(nativeConfig, version)
-        val schema = RealmInterop.realm_schema_new(tables)
+    private fun init() {
+        val internalPath = if (path == null || path.isEmpty()) {
+            val directory = PlatformHelper.appFilesDirectory()
+            // FIXME Proper platform agnostic file separator: File.separator is not available for Kotlin/Native
+            //  https://github.com/realm/realm-kotlin/issues/75
+            "$directory/$name.realm"
+        } else path
+
+        RealmInterop.realm_config_set_path(nativeConfig, internalPath)
+        RealmInterop.realm_config_set_schema_mode(
+            nativeConfig,
+            SchemaMode.RLM_SCHEMA_MODE_AUTOMATIC
+        )
+        RealmInterop.realm_config_set_schema_version(nativeConfig, version = 0)
+        val schema = RealmInterop.realm_schema_new(mapOfKClassWithCompanion.values.map { it.`$realm$schema`() })
         RealmInterop.realm_config_set_schema(nativeConfig, schema)
-    }
 
-    data class Builder(
```

Comment:
See my comment in the PR, but I don't think we should remove the Builder completely